src/engine/Operation.cpp

```diff
@@ -625,3 +625,24 @@ uint64_t Operation::getSizeEstimate() {
     return getSizeEstimateBeforeLimit();
   }
 }
+
+// _____________________________________________________________________________
+std::unique_ptr<Operation> Operation::clone() const {
+  auto result = cloneImpl();
+  AD_CORRECTNESS_CHECK(result->_executionContext == _executionContext);
+  auto areChildrenDifferent = [this, &result]() {
+    auto ownChildren = getChildren();
+    auto otherChildren = result->getChildren();
+    if (ownChildren.size() != otherChildren.size()) {
+      return false;
+    }
+    for (size_t i = 0; i < ownChildren.size(); i++) {
+      if (ownChildren.at(i) == otherChildren.at(i)) {
+        return false;
+      }
+    }
+    return true;
+  };
+  AD_CORRECTNESS_CHECK(areChildrenDifferent());
```

Comment:
Can also include expensive checks for the equality of variableToColumnMaps, sizeEstimates, multiplicities, whatNotElse

Reply:
(have a look at the public interface of `Operation` for inspiration.

Reply:
Many of these don't work because their public interface is not const (even though it seems like it could be const for most of them)